storagemarket/impl/client.go

```diff
@@ -348,14 +386,19 @@ func (c *clientDealEnvironment) Node() storagemarket.StorageClientNode {
 }
 
 func (c *clientDealEnvironment) WriteDealProposal(p peer.ID, proposalCid cid.Cid, proposal network.Proposal) error {
-	s, err := c.c.net.NewDealStream(p)
-	if err != nil {
-		return err
-	}
-	err = c.c.conns.AddStream(proposalCid, s)
+	s, err := c.c.conns.DealStream(proposalCid)
+
 	if err != nil {
-		return err
+		s, err = c.c.net.NewDealStream(p)
```

Comment:
small thing but maybe extract these six lines to a function -- you use them up in restart as well.